### PR TITLE
EOS-26962: Delete policy validation for user attached policy

### PR DESCRIPTION
**Pull request by soniyamoholkar**
# Problem Statement
- Add delete policy validation for attached user policy

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design
- https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/706805920/Low+Level+Design

# Coding
   Checklist for Author - done
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author - done
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM
- http://eos-jenkins.colo.seagate.com/job/S3server/job/S3Server-PreMerge-Test/829/ 

# Impact Analysis done
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist  done
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation done
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: auth/server/src/main/java/com/seagates3/controller/PolicyController.java [hunk 1]
```diff
@@ -143,6 +143,13 @@ public ServerResponse create() {
         LOGGER.error(requestBody.get("PolicyARN") + "Policy does not exists");
         return responseGenerator.noSuchEntity();
       }
+      if (policy.getAttachmentCount() > 0) {
+        LOGGER.error(
+            requestBody.get("PolicyARN") +
```
Comment (kupranay) [inline auth/server/src/main/java/com/seagates3/controller/PolicyController.java:148]: please use constants

File: auth/server/src/main/java/com/seagates3/controller/PolicyController.java [hunk 2]
```diff
@@ -143,6 +143,13 @@ public ServerResponse create() {
...
+            " cannot be deleted because it is attached to the entities.");
+        return responseGenerator.deletePolicyConflict(
+            "Cannot delete a policy attached to entities.");
```
Comment (kupranay) [inline auth/server/src/main/java/com/seagates3/controller/PolicyController.java:151]: it should be either "can not" or "can't"

**Review by kupranay — CHANGES_REQUESTED** [review]

**Review by bkirunge7 — APPROVED** [review]

File: auth/server/src/main/java/com/seagates3/controller/PolicyController.java (hunk 2 above)
Comment (soniyamoholkar) [reply auth/server/src/main/java/com/seagates3/controller/PolicyController.java:151]: @kupranay The error message is exactly copied from aws. So please let me know I still need to change.
image

File: auth/server/src/main/java/com/seagates3/controller/PolicyController.java (hunk 1 above)
Comment (soniyamoholkar) [reply auth/server/src/main/java/com/seagates3/controller/PolicyController.java]: Done. Please check

Comment (soniyamoholkar) [reply auth/server/src/main/java/com/seagates3/controller/PolicyController.java]: Done. Please check.

**Review by shrimetta — APPROVED** [review]

**Review by kupranay — APPROVED** [review]

**Pull request merged**